### PR TITLE
feature/REDBOX-282-azure-ai-api-end-point

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a User I want redbox to use an Azure instance of OpenAI so that my data is secure

## Relevant links

https://technologyprogramme.atlassian.net/browse/REDBOX-282

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/9289984838

### CODE REVIEW

File: infrastructure/aws/ecs.tf [hunk 1]
```diff
@@ -25,8 +25,11 @@ locals {
     "DJANGO_SETTINGS_MODULE" : "redbox_app.settings",
     "DEBUG" : true,
     "AWS_REGION" : var.region,
-    "OPENAI_API_KEY" : var.openai_api_key,
     "FROM_EMAIL" : var.from_email,
+    "LLM__TYPE": var.llm_type,
+    "LLM__NAME": var.llm_name,
+    "LLM__API_BASE": var.llm_api_base,
+    "AZURE_API_KEY": var.azure_api_key
```
Comment (lmwilkigov) [inline infrastructure/aws/ecs.tf:32]: Please change `AZURE_API_KEY` to `AZURE_OPENAI_API_KEY` and `LLM__API_BASE` to `AZURE_OPENAI_ENDPOINT` to match convention from Azure documentation: https://learn.microsoft.com/en-us/azure/ai-services/openai/gpt-v-quickstart?tabs=image%2Ccommand-line&pivots=programming-language-python#environment-variables

May seem nitpicky but some libraries look for this things in env (for example some eval libraries in notebooks)

**Review by lmwilkigov — CHANGES_REQUESTED** [review]

File: infrastructure/aws/ecs.tf (hunk 1 above)
Comment (gecBurton) [reply infrastructure/aws/ecs.tf:32]: sounds reasonable @lmwilkigov

Comment (gecBurton) [reply infrastructure/aws/ecs.tf]: [69ef416](https://github.com/i-dot-ai/redbox-copilot/pull/472/commits/69ef4168e8860e8bce385cb0cd7bd35416a2a43c)
[2cab6ba](https://github.com/i-dot-ai/redbox-copilot/pull/472/commits/2cab6baf941d61f885d6ea3e37caedc3d1588db3)

File: redbox/models/settings.py [hunk 2]
```diff
@@ -26,16 +30,27 @@ class ElasticCloudSettings(BaseModel):
     subscription_level: str = "basic"
 
 
+class LlmSettings(BaseModel):
+    azure_openai_endpoint: str | None = None
```
Comment (gecBurton) [inline redbox/models/settings.py:34]:
```suggestion
```

File: infrastructure/aws/variables.tf [hunk 3]
```diff
@@ -82,18 +82,42 @@ variable "image_tag" {
...
+variable "openai_model" {
+  type        = string
+  default     = "gpt-3.5-turbo"
+  description = "OPENAI model to use"
+}
+
 variable "openai_api_key" {
   type        = string
   sensitive   = true
+  default     = null
```
Comment (gecBurton) [inline infrastructure/aws/variables.tf:99]: this could now be null if we use azure

File: core_api/src/routes/chat.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import logging
+import os
 from http import HTTPStatus
 from typing import Annotated
 from uuid import UUID
 
-from fastapi import Depends, FastAPI, HTTPException, WebSocket
+from fastapi import Depends, FastAPI, HTTPException
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:7]:
```suggestion
from fastapi import Depends, FastAPI, HTTPException, WebSocket
```

File: core_api/src/routes/chat.py [hunk 5]
```diff
@@ -16,6 +17,7 @@
 from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
 from langchain_core.runnables import Runnable
 from langchain_elasticsearch import ApproxRetrievalStrategy, ElasticsearchStore
+from starlette.websockets import WebSocket
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:20]:
```suggestion
```

File: .github/workflows/integration.yml [hunk 6]
```diff
@@ -129,3 +132,7 @@ jobs:
       env:
         SLACK_WEBHOOK_URL: ${{ secrets.SLACK_NOTIFICATIONS_TOKEN }}
         SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+    - name: Dump logs
```
Comment (gecBurton) [inline .github/workflows/integration.yml:136]: this is useful for debugging

**Pull request merged**